tests/platform_tests/fwutil/fwutil_common.py

```diff
@@ -0,0 +1,228 @@
+import pytest
+import os
+import json
+import logging
+
+from copy import deepcopy
+
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+DEVICES_PATH="usr/share/sonic/device"
+TIMEOUT=3600
+REBOOT_TYPES = {
+        "cold": "reboot",
+        "warm": "warm-reboot",
+        "fast": "fast-reboot"
+        }
+
+def find_pattern(lines, pattern):
+    for line in lines:
+        if pattern.match(line):
+            return True
+    return False
+
+def power_cycle(duthost=None, pdu_ctrl=None, delay_time=60):
+    if pdu_ctrl is None:
+        pytest.skip("No PSU controller for %s, skipping" % duthost.hostname)
+
+    all_outlets = pdu_ctrl.get_outlet_status()
+
+    for outlet in all_outlets:
+        pdu_ctrl.turn_off_outlet(outlet)
+    time.sleep(delay_time)
+    for outlet in all_outlets:
+        pdu_ctrl.turn_on_outlet(outlet)
+
+def reboot(duthost, pdu_ctrl, reboot_type):
```

Comment:
@alexrallen why do we need to instantiate PDU controller in advance if there is a chance it won't be used?

Reply:
Its a fixture, I use the fixture in all test cases, I don't see any significant detriment to initializing it, the time it takes to init is insignificant and it simplifies the code by doing this. The test only skips if it both needs it and it is is not implemented for the given DUT.

Reply:
@alexrallen the overhead here in passing as a separate arg thru all the functions/methods, regardless the place where it's really needed

Reply:
Yes I agree I would rather not but it seems this module is intended to be used as a fixture and it could possibly be used in any test because a firmware update could require a power cycle. 

Front-loading the logic to try to prevent passing it would result in the code being much less organized than the pdu controller being passed along with the rest of the fixtures. 

If you know of another way to init it we can try that on-the-fly.

Reply:
@alexrallen there is a way to minimize the fixture list by using this pattern:
```python
    def __execute_task(self, request, cmd):
        localhost = request.getfixturevalue('localhost')
        duthost = request.getfixturevalue('duthost')
```
Since you decided to work with the pure fixture objects, you can either leave it as is or refactor. The decision is yours.

Reply:
Ah I see, thank you for informing me of this feature. I will leave as-is for the time being (if I were to want to move to this I would probably alter most of the function signatures), but I do see where this could be enhanced in the future using this.